[ADD] contract_sale_generation_payment_mode

victor-champonnois opened the pull request:
Set the payment mode from the contract on sales orders created by the contract.

victor-champonnois committed:
[ADD] contract_sale_generation_payment_mode

victor-champonnois committed:
[ADD] contract_sale_generation_payment_mode

huguesdk reviewed (approved):
lgtm! (amended some slight fixes to the commit.)

huguesdk commented:
@OCA/human-resources-maintainers could someone please review or merge this?

rousseldenis commented:
@victor-champonnois Thanks for this. Maybe a little test ? Thanks

victor-champonnois commented:
@rousseldenis thanks, I've added some tests

victor-champonnois commented:
I don't manage to see why the codecov stays red, an idea on how to fix this ? 
It states that these lines are not covered, but I don't see a way to cover them more than what I've already did. Can someone help ? :)

```
+         res = super()._prepare_recurring_sales_values(date_ref)
+         if res:
+             res[0]["payment_mode_id"] = self.payment_mode_id.id
+         return res
```

huguesdk committed:
[FIX] run tests at post_install

tests should be tagged as post_install, as ContractSaleCommon sets up a
coa if none has been loaded.

huguesdk committed:
[FIX] contract_sale_generation_payment_mode: run tests at post_install

tests should be tagged as post_install, as ContractSaleCommon sets up a
coa if none has been loaded.

huguesdk commented:
according to the logs and the coverage report, the test files are loaded, but the tests don’t run. those of `contract_sale_generation` don’t run either. this is due to a recent change in the tests of `contract_sale_generation` (in #1201). i fixed it in #1305 (and tagged the tests as `post_install` here also). i think the coverage will be fine as soon as #1305 is merged.

victor-champonnois commented:
many thanks @huguesdk !

huguesdk commented:
@OCA/project-service-maintainers could someone please merge this?

mamcode commented on contract_sale_generation_payment_mode/models/contract.py:
```diff
@@ -0,0 +1,15 @@
+        res = super()._prepare_recurring_sales_values(date_ref)
+        if res:
+            res[0]["payment_mode_id"] = self.payment_mode_id.id
+        return res
```
I think it's fine, but it seems more consistent with the module's design to put this code in the `_prepare_sale()` method. What do you think?

huguesdk replied:
@mamcode good catch, thanks! indeed. it is fixed now, and this fixes a bug where the payment mode would only be set on the first sale order if the method was called on a record set with multiple records.
